test/jdk/javax/management/notification/BroadcasterSupportDeadlockTest.java

```diff
@@ -105,31 +101,25 @@ public void run() {
         t2.setDaemon(true);
         t2.start();
 
-        /* Wait for Thread 2 to be blocked on the monitor or to
-           succeed.  */
+        /* Wait for Thread 2 to be blocked on the monitor or to succeed. */
         Thread.sleep(100);
 
-        for (int i = 0; i < 1000/*ms*/; i++) {
-            t2.join(1/*ms*/);
-            switch (t2.getState()) {
-            case TERMINATED:
+        for (int i = 0; i < 1000; i++) {
+            t2.join(100 /*ms*/);
+            if (t2.getState() == Thread.State.TERMINATED) {
                 System.out.println("TEST PASSED");
                 return;
-            case BLOCKED:
-                java.util.Map<Thread,StackTraceElement[]> traces =
-                    Thread.getAllStackTraces();
-                showStackTrace("Thread 1", traces.get(t1));
-                showStackTrace("Thread 2", traces.get(t2));
-                System.out.println("TEST FAILED: deadlock");
-                System.exit(1);
-                break;
-            default:
-                break;
             }
         }
 
-        System.out.println("TEST FAILED BUT DID NOT NOTICE DEADLOCK");
-        Thread.sleep(10000);
+        if (t2.getState() == Thread.State.BLOCKED) {
+            System.out.println("TEST FAILED: deadlock");
+        } else {
+            System.out.println("TEST FAILED BUT DID NOT NOTICE DEADLOCK (state = " + t2.getState() + ")");
+        }
+        java.util.Map<Thread,StackTraceElement[]> traces = Thread.getAllStackTraces();
+        showStackTrace("Thread 1", traces.get(t1));
+        showStackTrace("Thread 2", traces.get(t2));
```

Comment:
Could you please replace System.exit() with throwing Exception. Other looks good.

Reply:
OK sure.
FYI I count 137 System.exit calls in test/jdk/javax/management
That's after I take out the existing 3 that are in this test. 8-)